oauth2client/service_account.py

```diff
@@ -461,3 +465,100 @@ def create_delegated(self, sub):
+                the payload of the JWT. 
+            Returns:
+                An AccessTokenInfo with the signed jwt
+        """
+        if additional_claims is not None:
+          # Create a 1 time token
+          return self._create_token(additional_claims)
+        else:
+          if self.access_token is None or self.access_token_expired:
+            self._refresh(None)
+
+        return AccessTokenInfo(access_token=self.access_token,
+                              expires_in=self._expires_in())
+    def revoke(self, http):
+        # TODO(kpayson): Is there any way to actually invalidate a signed JWT?
+        raise TokenRevokeError('Cannot revoke JWTAccessCredentials')
+
+    @classmethod
+    def from_json(cls, json_data):
+      if not isinstance(json_data, dict):
+        json_data = json.loads(_from_bytes(json_data))
+
+      service_account_creds = ServiceAccountCredentials.from_json(json_data['_creds'])
+      retval = cls(service_account_creds, 
+                   additional_claims=json_data['_claims'],
+                   access_token=json_data['access_token'],
+                   token_expiry=json_data['token_expiry'])
+      return retval
+       
+    def _to_json(self, strip, to_serialize=None):
+      if to_serialize is None:
+            to_serialize = copy.copy(self.__dict__)
+      creds_val = to_serialize.get('_creds')
+      if creds_val is not None:
+        to_serialize['_creds'] = json.loads(self._creds.to_json())
+      return super(JWTAccessCredentials, self)._to_json(
+        strip, to_serialize=to_serialize)
+
+    def _refresh(self, http):
+        self.token_expiry = _UTCNOW() + datetime.timedelta(seconds=self.MAX_TOKEN_LIFETIME_SECS)
+        self.access_token = self._create_token()
+    
+    def _create_token(self, additional_claims=None):
+        now = time.time()
+        payload = {
+            'iat': now,
+            'exp': now + self.MAX_TOKEN_LIFETIME_SECS,
+            'iss': self._creds._service_account_email,
+            'sub': self._creds._service_account_email
+        }
+        payload.update(self._claims)
+        if additional_claims is not None:
+          payload.update(additional_claims)
+        return crypt.make_signed_jwt(self._creds._signer, payload,
+                                     key_id=self._creds._private_key_id)
+
+    @classmethod
+    def create_with_claims(cls, service_account_creds, claims):
+        """Create a self-signed JWT credentials from service account credentials
+
+        Args:
+            service_account_creds: ServiceAccountCredentials, the
+              credentials used to sign the JWTs
+            claims: dict, the claims to place in the JWT payload.
+              The 'aud' claim must be specified to authenticate with Google. 
+              Ex.  {'aud' : {https://googleapis/MyService'}
+
+        Returns:
+            SelfSignedJWTCredentials for the service account
+        """
+        return cls(service_account_creds, claims)
```

Comment:
This should merge the current instance's claims with the new claims and return an instance.